src/dotnet/UI.Blazor/Components/Modal/BlazoredModal/BlazoredModal.razor

```diff
@@ -0,0 +1,165 @@
+﻿@using System.Collections.ObjectModel
+@using Microsoft.AspNetCore.Components.Routing
+@namespace Blazored.Modal
+
+@inject NavigationManager NavigationManager
+@inject IJSRuntime JsRuntime
+@implements IAsyncDisposable
+
+<CascadingValue Value="this">
+    <CascadingValue Value="_globalModalOptions">
+        @foreach (var modal in _modals)
+        {
+            @modal.ModalInstance
+        }
+    </CascadingValue>
+</CascadingValue>
+
+@code {
+    [CascadingParameter] private IModalService CascadedModalService { get; set; } = default!;
```

Comment:
- Curious why it is cascaded... I.e. why not injected?
- Why its name isn't just "ModalService"?

Reply:
fixed